cfme/tests/infrastructure/test_snapshot.py

```diff
@@ -41,40 +43,52 @@ def domain(request):
     return dom
 
 
-@pytest.fixture(scope="module")
-def test_vm(setup_provider_modscope, provider, vm_name, request):
-    """Fixture to provision appliance to the provider being tested if necessary"""
-    vm = VM.factory(vm_name, provider, template_name=provider.data['full_template']['name'])
+def provision_vm(provider, vm_name, request, small_template=True):
+    if small_template:
+        vm = VM.factory(vm_name, provider, template_name=provider.data['small_template'])
```

Comment:
@psav @apagac There is inconsistency in the template naming throughout providers in cfme_data.yaml, which has been recorded in an issue, and that I have an open card to address. I'll be adding a 'templates' section to each provider, with consistent structure for `small_template`, `full_template` arrays that define `name` and `creds` for each template.

At that time I'll be auditing tests to update their access of the templates in the yaml, and can collapse this to a consistent access path at that time.